src/spotify_to_tidal/sync.py

```diff
@@ -270,11 +270,12 @@ def sync_playlist(spotify_session: spotipy.Spotify, tidal_session: tidalapi.Sess
         return
 
     task_description = "Searching Tidal for {}/{} tracks in Spotify playlist '{}'".format(len(spotify_tracks) - cache_hits, len(spotify_tracks), spotify_playlist['name'])
-    tidal_tracks = call_async_with_progress(tidal_search, spotify_tracks, task_description, config.get('subprocesses', 25), tidal_session=tidal_session)
+    tidal_tracks = call_async_with_progress(tidal_search, spotify_track_mappings, task_description, config.get('subprocesses', 25), tidal_session=tidal_session)
     for index, tidal_track in enumerate(tidal_tracks):
-        spotify_track = spotify_tracks[index][0]
+        spotify_track = spotify_tracks[index]
         if tidal_track:
-            tidal_track_ids.append(tidal_track)
+            track_id = tidal_track.id if isinstance(tidal_track, tidalapi.Track) else tidal_track
```

Comment:
Something happened with the list, sometimes its an int sometimes its a tidal track object. Need to look further